ssvsigner/web3signer/web3signer.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		logger.Error("failed to send http request", zap.Error(err))
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+	defer func() {
+		if err := httpResp.Body.Close(); err != nil {
+			logger.Error("failed to close http response body", zap.Error(err))
+		}
+	}()
+
+	respBytes, err := io.ReadAll(httpResp.Body)
+	if err != nil {
+		logger.Error("failed to read http response body", zap.Error(err))
+		return nil, fmt.Errorf("read response body: %w", err)
+	}
+	var resp DeleteKeystoreResponse
+	if err := json.Unmarshal(respBytes, &resp); err != nil {
+		logger.Error("failed to unmarshal http response body", zap.String("body", string(respBytes)), zap.Error(err))
+	}
+
+	if httpResp.StatusCode != http.StatusOK {
+		logger.Error("failed to delete keystore",
+			zap.Int("status_code", httpResp.StatusCode),
+			zap.String("message", resp.Message))
+		return nil, fmt.Errorf("unexpected status %d: %v", httpResp.StatusCode, resp.Message)
+	}
+
+	logger.Info("deleted keystores", zap.Any("response", string(respBytes)))
+
+	var statuses []Status
+	for _, data := range resp.Data {
+		statuses = append(statuses, data.Status)
+	}
+
+	return statuses, nil
+}
+
+// Sign signs using https://consensys.github.io/web3signer/web3signer-eth2.html#tag/Signing/operation/ETH2_SIGN
+func (c *Web3Signer) Sign(sharePubKey []byte, payload SignRequest) ([]byte, error) {
+	sharePubKeyHex := "0x" + hex.EncodeToString(sharePubKey)
+	logger := c.logger.With(
+		zap.String("request", "Sign"),
+		zap.String("share_pubkey", sharePubKeyHex),
+		zap.String("type", string(payload.Type)),
+	)
+	logger.Info("signing")
+
+	body, err := json.Marshal(payload)
+	if err != nil {
+		logger.Error("failed to marshal payload", zap.Error(err))
+		return nil, fmt.Errorf("marshal payload: %w", err)
+	}
+
+	url := fmt.Sprintf("%s/api/v1/eth2/sign/%s", c.baseURL, sharePubKeyHex)
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewReader(body))
+	if err != nil {
+		logger.Error("failed to create http request", zap.Error(err))
+		return nil, fmt.Errorf("create sign request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		logger.Error("failed to send http request", zap.Error(err))
+		return nil, fmt.Errorf("sign request failed: %w", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			logger.Error("failed to close http response body", zap.Error(err))
+		}
+	}()
+
+	respData, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logger.Error("failed to read http response body", zap.Error(err))
+		return nil, fmt.Errorf("read response body: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		logger.Error("sign request failed",
+			zap.Int("status_code", resp.StatusCode),
+			zap.Any("response", string(respData)),
+			zap.Any("request", string(body)),
+			zap.Any("url", url))
+		return nil, fmt.Errorf("unexpected status %d: %v", resp.StatusCode, respData)
+	}
```

Comment:
Ditto – looks like the same issue as in import and delete. Should we handle error responses earlier instead of passing them to the caller?

Reply:
For signing, `web3signer` never returns an error if it responds with `http.StatusOK`: https://consensys.github.io/web3signer/web3signer-eth2.html#tag/Signing/operation/ETH2_SIGN